### ISSUE TITLE
Task Assignee Reward Crash

### ISSUE THREAD

**Issue by pavlovcik** [ubiquity/ubiquibot#687]
I skimmed the logs and the only thing that stood out to me is log id `17061`

> Penalty removed

Perhaps there is a bug when the contributor has a "debt" and then the reward amount must subtract the debt.

Logs from the last 15 minutes: 
[supabase_iyybhhiflwbsjopsgaow_Get all results for issue 663 in the logs table (1).csv](https://github.com/ubiquity/ubiquity-dollar/files/12462035/supabase_iyybhhiflwbsjopsgaow_Get.all.results.for.issue.663.in.the.logs.table.1.csv)

> Error: Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="symbol()", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.7.0)

_Originally posted by @ubiquibot[bot] in https://github.com/ubiquity/ubiquity-dollar/issues/663#issuecomment-1696985103_

**Comment by molecula451** [comment]
@rndquu forward this to pay-ubq.fi repo as the bug it's there around [render-token.symbol.ts](https://github.com/ubiquity/pay.ubq.fi/blob/development/static/scripts/rewards/render-transaction/render-token-symbol.ts)

```javascript
export async function renderTokenSymbol({ table, requestedAmountElement }: { table: Element; requestedAmountElement: Element }): Promise<void> {
  const contract = await getDaiLikeContract(app.txData.permit.permitted.token);
  const symbol = await contract.symbol();
  table.setAttribute(`data-contract-loaded`, "true");
  requestedAmountElement.innerHTML = `<a target="_blank" rel="noopener noreferrer" href="${app.explorerUrl}/token/${app.txData.permit.permitted.token}?a=${
    app.txData.owner
  }">${Number(app.txData.transferDetails.requestedAmount) / 1000000000000000000} ${symbol}</a>`;
}
```

**Comment by rndquu** [comment]
> @rndquu forward this to pay-ubq.fi repo as the bug it's there around [render-token.symbol.ts](https://github.com/ubiquity/pay.ubq.fi/blob/development/static/scripts/rewards/render-transaction/render-token-symbol.ts)
> 
> ```js
> export async function renderTokenSymbol({ table, requestedAmountElement }: { table: Element; requestedAmountElement: Element }): Promise<void> {
>   const contract = await getDaiLikeContract(app.txData.permit.permitted.token);
>   const symbol = await contract.symbol();
>   table.setAttribute(`data-contract-loaded`, "true");
>   requestedAmountElement.innerHTML = `<a target="_blank" rel="noopener noreferrer" href="${app.explorerUrl}/token/${app.txData.permit.permitted.token}?a=${
>     app.txData.owner
>   }">${Number(app.txData.transferDetails.requestedAmount) / 1000000000000000000} ${symbol}</a>`;
> }
> ```

The **bot** [responded](https://github.com/ubiquity/ubiquity-dollar/issues/663#issuecomment-1696985103) with error, it can't be a pay.ubq.fi related bug

**Comment by pavlovcik** [comment]
> lmao, right, so it's over [here ](https://github.com/ubiquity/ubiquibot/blob/5ef5f446fda1497af9a5139afa453845f91b71cb/src/helpers/contracts.ts#L4) leaving it here for ref

Assuming the request is timing out, I suppose we can wrap it in a try catch? Seems sloppy but I'm not sure what else could be the solution.

**Comment by molecula451** [comment]
something similar, a potential solution would be` ${tokenSymbol?.toUpperCase()} ` type `?` chaining the `tokenSymbol` var, but need test on the playground

**Comment by pavlovcik** [comment]
If the token symbol fails to load, we can fallback to the full token name. If that fails, we can fallback to "Tokens". This is in the context of the rendered claim button e.g. **[ CLAIM 100 TOKENS ]**

**Comment by molecula451** [comment]
> If the token symbol fails to load, we can fallback to the full token name. If that fails, we can fallback to "Tokens". 

yes, I agree seems a proper solution, so the bot ignores the call revert by the contract call

>  This is in the context of the rendered claim button

yes
